src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -857,16 +856,9 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
 
         case tpnme.RuntimeAnnotationATTR =>
           val numAnnots = u2()
-          val annots = new ListBuffer[AnnotationInfo]
           numAnnots times {
-            parseAnnotation(u2()).foreach(annots.addOne)
+            parseAnnotation(u2()).foreach(addUniqueAnnotation(sym, _))
```

Comment:
Should we restrict to `@Deprecated`, avoid forcing existing symbol annotations for other cases?

Reply:
Indeed, I wondered if only deprecation has this behavior but did not follow up.

Reply:
Deprecated is the only "predefined attribute" that becomes an AnnotationInfo. The check is not name-based because we have a symbol.